hbase_sql/src/main/java/com/splicemachine/mrio/api/core/AbstractSMInputFormat.java

```diff
@@ -191,6 +192,39 @@
         }
     }
 
+    /**
+     * Checks the sequence of split rows. If any gap is found, the splitting has to be recalculated again.
+     *
+     * @param lss
+     * @return
+     */
+    protected boolean isRefreshNeeded(List<InputSplit> lss) {
+        if (lss.size() < 2 || !lss.stream().allMatch(is -> (is instanceof SMSplit))) {
```

Comment:
Do we ever get something other than SMSplit? Is it just defensive programming?

Reply:
I've changed it to assert the instanceof